### PR TITLE
Fix disk I/O error when resetting short-term memory.

### LINKED ISSUE #1723: [BUG] error resetting short-term memory
### Description

A disk I/O error occurs when resetting short-term memory

### Steps to Reproduce

1. Create new crewai crew
2. Run `crewai reset-memories -s` or `crewai reset-memories -a`

### Expected behavior

No errors should be raised and a successful memory reset message should be displayed

### Screenshots/Code snippets

```bash
$ crewai reset-memories -a                                                          
An unexpected error occurred: An error occurred while resetting the short-term memory: An error occurred while resetting the short_term memory: disk I/O error

$ crewai reset-memories -s
An unexpected error occurred: An error occurred while resetting the short-term memory: An error occurred while resetting the short_term memory: disk I/O error
```

Long term memory reset is fine:

```bash
$ crewai reset-memories -l
Long term memory has been reset.
```

### Operating System

Other (specify in additional context)

### Python Version

3.12

### crewAI Version

0.86.0

### crewAI Tools Version

0.17.0

### Virtual Environment

Venv

### Evidence

pyproject_toml

### Possible Solution

Inside the `reset` method on `RAGStorage`, call `self.app.reset()` before calling manually removing the folder with `shutil.rmtree(f"{db_storage_path()}/{self.type}")`

### Additional context

None

### PR DESCRIPTION
**Pull request by caike**
Reset chromadb client and nullifies references before removing directory.

Fixes #1723

### CODE REVIEW

**Comment by joaomdmoura** [comment]
Disclaimer: This review was made by a crew of AI Agents.

# Code Review for PR #1724

## Overview
This pull request addresses a disk I/O error encountered when resetting short-term memory in the `rag_storage.py` file. The primary change involves reordering operations and ensuring proper cleanup of references to mitigate such errors.

## Positive Aspects
- The changes effectively solve the root cause of the disk I/O issue by resetting the ChromaDB client before removing the associated directories.
- Object references are appropriately nullified after cleanup, which is crucial for preventing memory leaks.
- Existing error handling for readonly database cases has been preserved, ensuring stability under specific failure conditions.

## Areas for Improvement

### 1. Resource Cleanup Order
#### Current Implementation
```python
def reset(self) -> None:
    try:
        if self.app:
            self.app.reset()
            shutil.rmtree(f"{db_storage_path()}/{self.type}")
            self.app = None
            self.collection = None
    except Exception as e:
        if "attempt to write a readonly database" in str(e):
            # Ignore this specific error
```

#### Recommended Implementation
It is recommended to adjust the order of operations to:

- Reset the ChromaDB client first.
- Clear references before performing file system operations.
- Perform cleanup operations as a final step.

### 2. Enhanced Error Handling
#### Current Strategy
The current strategy only accounts for readonly database errors. 

#### Recommended Addition
- Expand error handling to log exceptions more thoroughly for better traceability and debugging.
- Implement handling for other potential exceptions during the reset and cleanup processes.

### Additional Suggestions
1. **Logging Enhancements:** Integrate logging statements throughout the `reset` method to capture critical operation states, especially before and after significant actions.
  
2. **Pre-check Method:** Introduce a utility method to check if cleanup is required before proceeding with the reset function to avoid unnecessary operations.

3. **Documentation:** Improve documentation for the `reset` method, outlining the overall process, potential pitfalls, and how errors are handled.

4. **Testing:** Ensure robust test coverage, including:
   - Unit tests verifying success under various conditions.
   - Tests for edge cases such as readonly database errors or missing directories.
   - Integration tests to validate resource management and cleaning behavior.

5. **Explore "Soft Reset":** Consider implementing a soft reset option that clears only data while keeping the client functional, which could improve user experience during resets.

## Conclusion
The changes submitted in this PR effectively mitigate the disk I/O issue by ensuring appropriate order and cleanup in resource management. The recommendations aim to strengthen the robustness and maintainability of the code while enhancing debuggability. Overall, these improvements will significantly benefit the performance and stability of memory storage management in the ChromaDB environment.

File: src/crewai/memory/storage/rag_storage.py [hunk 1]
```diff
@@ -150,9 +150,11 @@ def _generate_embedding(self, text: str, metadata: Dict[str, Any]) -> None:  # t
 
     def reset(self) -> None:
         try:
-            shutil.rmtree(f"{db_storage_path()}/{self.type}")
             if self.app:
                 self.app.reset()
+                self.app = None
+                self.collection = None
+                shutil.rmtree(f"{db_storage_path()}/{self.type}")
```
Comment (lorenzejay) [inline src/crewai/memory/storage/rag_storage.py]: Awesome work

**Review by lorenzejay — APPROVED** [review]
Approved ✅

File: src/crewai/memory/storage/rag_storage.py (hunk 1 above)
Comment (lorenzejay) [inline src/crewai/memory/storage/rag_storage.py]: I like this comment: 

```python
def reset(self) -> None:
    try:
        if self.app:
            self.app.reset()
            shutil.rmtree(f"{db_storage_path()}/{self.type}")
            self.app = None
            self.collection = None
    except Exception as e:
        if "attempt to write a readonly database" in str(e):
            # Ignore this specific error
```

Comment (lorenzejay) [reply src/crewai/memory/storage/rag_storage.py]: This adds cleanup after the rm of the directories

File: src/crewai/memory/storage/rag_storage.py [hunk 2]
```diff
@@ -150,9 +150,11 @@ def _generate_embedding(self, text: str, metadata: Dict[str, Any]) -> None:  # t
 
     def reset(self) -> None:
         try:
-            shutil.rmtree(f"{db_storage_path()}/{self.type}")
             if self.app:
                 self.app.reset()
+                shutil.rmtree(f"{db_storage_path()}/{self.type}")
```
Comment (lorenzejay) [inline src/crewai/memory/storage/rag_storage.py:155]: Great work!

**Review by lorenzejay — APPROVED** [review]

**Pull request merged**